## Final Project Submission

Please fill out:
* Student name: Whitney Ndivi Ngili
* Student pace: full time
* Scheduled project review date/time: 
* Instructor name: Mark Tiba 
* Blog post URL:


In [122]:
# Your code here - remember to use markdown cells for comments as well!
import pandas as pd
import matplotlib.pyplot as plt
import sqlite3
import numpy as np

In [82]:
conn = sqlite3.connect("zippedData/im.db")

In [83]:
imdb_movies = pd.read_sql("""
SELECT name TableNames
FROM sqlite_master
WHERE type = 'table'
;
""",conn)
imdb_movies

TableNames
0   movie_basics
1      directors
2      known_for
3     movie_akas
4  movie_ratings
5        persons
6     principals
7        writers

In [86]:
bom_movies = pd.read_csv("zippedData/bom.movie_gross.csv.gz")
bom_movies.head()

title studio  domestic_gross  \
0                                  Toy Story 3     BV     415000000.0   
1                   Alice in Wonderland (2010)     BV     334200000.0   
2  Harry Potter and the Deathly Hallows Part 1     WB     296000000.0   
3                                    Inception     WB     292600000.0   
4                          Shrek Forever After   P/DW     238700000.0   

  foreign_gross  year  
0     652000000  2010  
1     691300000  2010  
2     664300000  2010  
3     535700000  2010  
4     513900000  2010

In [87]:
bom_movies.tail(10)

title      studio  domestic_gross foreign_gross  \
3377               Let Yourself Go        Men.         12200.0           NaN   
3378                 Hannah (2018)         PDF         11700.0           NaN   
3379                      Souvenir      Strand         11400.0           NaN   
3380  Furious (Legend of Kolovrat)     CARUSEL         10000.0           NaN   
3381           Beauty and the Dogs       Osci.          8900.0           NaN   
3382                     The Quake       Magn.          6200.0           NaN   
3383   Edward II (2018 re-release)          FM          4800.0           NaN   
3384                      El Pacto        Sony          2500.0           NaN   
3385                      The Swan  Synergetic          2400.0           NaN   
3386             An Actor Prepares       Grav.          1700.0           NaN   

      year  
3377  2018  
3378  2018  
3379  2018  
3380  2018  
3381  2018  
3382  2018  
3383  2018  
3384  2018  
3385  2018  
3386  2018

In [88]:
rt_movies = pd.read_csv('zippedData/rt.movie_info.tsv.gz',delimiter = '\t')
rt_movies.head()

id                                           synopsis rating  \
0   1  This gritty, fast-paced, and innovative police...      R   
1   3  New York City, not-too-distant-future: Eric Pa...      R   
2   5  Illeana Douglas delivers a superb performance ...      R   
3   6  Michael Douglas runs afoul of a treacherous su...      R   
4   7                                                NaN     NR   

                                 genre          director  \
0  Action and Adventure|Classics|Drama  William Friedkin   
1    Drama|Science Fiction and Fantasy  David Cronenberg   
2    Drama|Musical and Performing Arts    Allison Anders   
3           Drama|Mystery and Suspense    Barry Levinson   
4                        Drama|Romance    Rodney Bennett   

                            writer  theater_date      dvd_date currency  \
0                   Ernest Tidyman   Oct 9, 1971  Sep 25, 2001      NaN   
1     David Cronenberg|Don DeLillo  Aug 17, 2012   Jan 1, 2013        $   
2                   Allison Anders  Sep 13, 1996  Apr 18, 2000      NaN   
3  Paul Attanasio|Michael Crichton   Dec 9, 1994  Aug 27, 1997      NaN   
4                     Giles Cooper           NaN           NaN      NaN   

  box_office      runtime             studio  
0        NaN  104 minutes                NaN  
1    600,000  108 minutes  Entertainment One  
2        NaN  116 minutes                NaN  
3        NaN  128 minutes                NaN  
4        NaN  200 minutes                NaN

In [89]:
rt_reviews = pd.read_csv('zippedData/rt.reviews.tsv.gz',delimiter = "\t",encoding='latin-1')
rt_reviews.head()

id                                             review rating   fresh  \
0   3  A distinctly gallows take on contemporary fina...    3/5   fresh   
1   3  It's an allegory in search of a meaning that n...    NaN  rotten   
2   3  ... life lived in a bubble in financial dealin...    NaN   fresh   
3   3  Continuing along a line introduced in last yea...    NaN   fresh   
4   3             ... a perverse twist on neorealism...     NaN   fresh   

           critic  top_critic         publisher               date  
0      PJ Nabarro           0   Patrick Nabarro  November 10, 2018  
1  Annalee Newitz           0           io9.com       May 23, 2018  
2    Sean Axmaker           0  Stream on Demand    January 4, 2018  
3   Daniel Kasman           0              MUBI  November 16, 2017  
4             NaN           0      Cinema Scope   October 12, 2017

In [90]:
tmdb_movies = pd.read_csv('zippedData/tmdb.movies.csv.gz',index_col = 0)
tmdb_movies.head()

genre_ids     id original_language  \
0      [12, 14, 10751]  12444                en   
1  [14, 12, 16, 10751]  10191                en   
2        [12, 28, 878]  10138                en   
3      [16, 35, 10751]    862                en   
4        [28, 878, 12]  27205                en   

                                 original_title  popularity release_date  \
0  Harry Potter and the Deathly Hallows: Part 1      33.533   2010-11-19   
1                      How to Train Your Dragon      28.734   2010-03-26   
2                                    Iron Man 2      28.515   2010-05-07   
3                                     Toy Story      28.005   1995-11-22   
4                                     Inception      27.920   2010-07-16   

                                          title  vote_average  vote_count  
0  Harry Potter and the Deathly Hallows: Part 1           7.7       10788  
1                      How to Train Your Dragon           7.7        7610  
2                                    Iron Man 2           6.8       12368  
3                                     Toy Story           7.9       10174  
4                                     Inception           8.3       22186

In [91]:
tn_movies = pd.read_csv('zippedData/tn.movie_budgets.csv.gz')
tn_movies.head()

id  release_date                                        movie  \
0   1  Dec 18, 2009                                       Avatar   
1   2  May 20, 2011  Pirates of the Caribbean: On Stranger Tides   
2   3   Jun 7, 2019                                 Dark Phoenix   
3   4   May 1, 2015                      Avengers: Age of Ultron   
4   5  Dec 15, 2017            Star Wars Ep. VIII: The Last Jedi   

  production_budget domestic_gross worldwide_gross  
0      $425,000,000   $760,507,625  $2,776,345,279  
1      $410,600,000   $241,063,875  $1,045,663,875  
2      $350,000,000    $42,762,350    $149,762,350  
3      $330,600,000   $459,005,868  $1,403,013,963  
4      $317,000,000   $620,181,382  $1,316,721,747

Assessing imdb movies

In [176]:
q1 = pd.read_sql("""
SELECT *
FROM movie_ratings
;
""",conn)
q1

movie_id  averagerating  numvotes
0      tt10356526            8.3        31
1      tt10384606            8.9       559
2       tt1042974            6.4        20
3       tt1043726            4.2     50352
4       tt1060240            6.5        21
...           ...            ...       ...
73851   tt9805820            8.1        25
73852   tt9844256            7.5        24
73853   tt9851050            4.7        14
73854   tt9886934            7.0         5
73855   tt9894098            6.3       128

[73856 rows x 3 columns]

In [175]:
q2 = pd.read_sql("""
SELECT *
FROM movie_basics
;
""",conn)
q2

movie_id                                primary_title  \
0       tt0063540                                    Sunghursh   
1       tt0066787              One Day Before the Rainy Season   
2       tt0069049                   The Other Side of the Wind   
3       tt0069204                              Sabse Bada Sukh   
4       tt0100275                     The Wandering Soap Opera   
...           ...                                          ...   
146139  tt9916538                          Kuambil Lagi Hatiku   
146140  tt9916622  Rodolpho Teóphilo - O Legado de um Pioneiro   
146141  tt9916706                              Dankyavar Danka   
146142  tt9916730                                       6 Gunn   
146143  tt9916754               Chico Albuquerque - Revelações   

                                     original_title  start_year  \
0                                         Sunghursh        2013   
1                                   Ashad Ka Ek Din        2019   
2                        The Other Side of the Wind        2018   
3                                   Sabse Bada Sukh        2018   
4                             La Telenovela Errante        2017   
...                                             ...         ...   
146139                          Kuambil Lagi Hatiku        2019   
146140  Rodolpho Teóphilo - O Legado de um Pioneiro        2015   
146141                              Dankyavar Danka        2013   
146142                                       6 Gunn        2017   
146143               Chico Albuquerque - Revelações        2013   

        runtime_minutes                genres  
0                 175.0    Action,Crime,Drama  
1                 114.0       Biography,Drama  
2                 122.0                 Drama  
3                   NaN          Comedy,Drama  
4                  80.0  Comedy,Drama,Fantasy  
...                 ...                   ...  
146139            123.0                 Drama  
146140              NaN           Documentary  
146141              NaN                Comedy  
146142            116.0                  None  
146143              NaN           Documentary  

[146144 rows x 6 columns]

We can join the movie_basics and movie_ratings tables

In [178]:
q3 = pd.read_sql("""
SELECT *
FROM movie_basics b
JOIN movie_ratings r
USING (movie_id)
LIMIT 10
;
""",conn)
q3

movie_id                       primary_title  \
0  tt0063540                           Sunghursh   
1  tt0066787     One Day Before the Rainy Season   
2  tt0069049          The Other Side of the Wind   
3  tt0069204                     Sabse Bada Sukh   
4  tt0100275            The Wandering Soap Opera   
5  tt0112502                             Bigfoot   
6  tt0137204                     Joe Finds Grace   
7  tt0146592                         Pál Adrienn   
8  tt0154039                So Much for Justice!   
9  tt0159369  Cooper and Hemingway: The True Gen   

                       original_title  start_year  runtime_minutes  \
0                           Sunghursh        2013            175.0   
1                     Ashad Ka Ek Din        2019            114.0   
2          The Other Side of the Wind        2018            122.0   
3                     Sabse Bada Sukh        2018              NaN   
4               La Telenovela Errante        2017             80.0   
5                             Bigfoot        2017              NaN   
6                     Joe Finds Grace        2017             83.0   
7                         Pál Adrienn        2010            136.0   
8                      Oda az igazság        2010            100.0   
9  Cooper and Hemingway: The True Gen        2013            180.0   

                       genres  averagerating  numvotes  
0          Action,Crime,Drama            7.0        77  
1             Biography,Drama            7.2        43  
2                       Drama            6.9      4517  
3                Comedy,Drama            6.1        13  
4        Comedy,Drama,Fantasy            6.5       119  
5             Horror,Thriller            4.1        32  
6  Adventure,Animation,Comedy            8.1       263  
7                       Drama            6.8       451  
8                     History            4.6        64  
9                 Documentary            7.6        53

Assessing bom_movies

In [92]:
bom_movies.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 3387 entries, 0 to 3386
Data columns (total 5 columns):
 #   Column          Non-Null Count  Dtype  
---  ------          --------------  -----  
 0   title           3387 non-null   object 
 1   studio          3382 non-null   object 
 2   domestic_gross  3359 non-null   float64
 3   foreign_gross   2037 non-null   object 
 4   year            3387 non-null   int64  
dtypes: float64(1), int64(1), object(3)
memory usage: 132.4+ KB


1. foreign_gross dtype is object-change to int
2. assess null values per column:

In [93]:
bom_movies.isnull().sum()

title                0
studio               5
domestic_gross      28
foreign_gross     1350
year                 0
dtype: int64

In [117]:
#defining a function that prints the percentage of null rows in all columns 
def find_percentage_of_nulls(df):
    for column in df.columns:
        percentage = round(df[column].isnull().sum()/(len(df))*100,3)
        print(f"The percentage of null rows in column {column} is {percentage}%")

find_percentage_of_nulls(bom_movies)        

The percentage of null rows in column title is 0.0%
The percentage of null rows in column studio is 0.148%
The percentage of null rows in column domestic_gross is 0.827%
The percentage of null rows in column foreign_gross is 39.858%
The percentage of null rows in column year is 0.0%


- The studio column only has 5 rows with null values while the domestic_gross column only has 28. This represents 0.827% and 0.148% of the dataset respectively. It makes sense to drop them as they will not affect the data analysis.
- The foreign_gross column has almost 40% of the data being null. 

In [118]:
#dropping null rows in studion column
bom_movies.dropna(inplace = True, subset = ['studio','domestic_gross'])

In [134]:
bom_movies.foreign_gross.fillna('0',inplace = True)

In [166]:
bom_movies.foreign_gross = bom_movies.foreign_gross.map(lambda x:x.replace(",",""))

In [169]:
bom_movies.foreign_gross = bom_movies.foreign_gross.astype(float)

In [171]:
bom_movies.foreign_gross.mean()

45325179.49240167